### DISCUSSION TITLE
`Circle Size`'s max Value in `osu` ruleset should limited to 7.0

### DISCUSSION THREAD

**Discussion by cdwcgt** [ppy/osu#19391] [Q&A]
Circle size in stable editor can only set to 7.0 but lazer editor can set to 10.0

My point is that it can be done according to different sets of rules (set it can be override) like:
But I feel that the code structure will be changed if try to set this virtual, I don't know if it is feasible
May need to maintain the status quo temporarily?

Related code:
https://github.com/ppy/osu/blob/master/osu.Game/Screens/Edit/Setup/DifficultySection.cs

Related commit:
https://github.com/ppy/osu/commit/477de1bab0483a6b3e22b82a7fc30eaf4c9610aa

Related issues:
https://github.com/ppy/osu/issues/11049

**Comment by frenzibyte** [comment]
Well, you're proposing to revert what #11049 has changed, so there definitely needs to be a discussion before talking about the code.

**Reply by cdwcgt** [reply]
Recovery may not be necessary, it's more like each ruleset needs a different limit
like override this value via [OsuSetupSection.cs](https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Osu/Edit/Setup/OsuSetupSection.cs)
but [OsuSetupSection.cs](https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Osu/Edit/Setup/OsuSetupSection.cs) is not direct inheritance [DifficultySection.cs](https://github.com/ppy/osu/blob/master/osu.Game/Screens/Edit/Setup/DifficultySection.cs) so I'm worried it might be a big change

**Comment by smoogipoo** [comment]
This is intentional. Stable actually supports 0-10 if you edit the `.osu` file manually. The rest is up to the ranking criteria to decide.

**Comment by peppy** [comment]
An eventual plan may be to have a checkbox which allows overriding the rankable limits. I think it's good to offer only the rankable range by default, but it's a bit more work.
